addons/project/models/project.py

```diff
@@ -1163,10 +1273,31 @@ def default_get(self, default_fields):
 
         return vals
 
+    def read(self, fields=None, load='_classic_read'):
+        if fields and self.env.user.has_group('base.group_portal'):
+            for key in fields:
+                if not (key in self.SELF_READABLE_FIELDS or key.startswith('context_')):
+                    raise AccessError(_('You cannot read %s field in task.') % key)
+        return super(Task, self).read(fields=fields, load=load)
+
+    @api.model
+    def _ensure_portal_user_can_write(self, fields):
+        for field in fields:
+            if not (field in self.SELF_WRITABLE_FIELDS):
+                raise AccessError(_('You have not write access of %s field.') % field)
+
     @api.model_create_multi
     def create(self, vals_list):
+        is_portal_user = self.env.user.has_group('base.group_portal')
         default_stage = dict()
         for vals in vals_list:
+            portal_can_create = False
+            if is_portal_user:
+                self.check_access_rights('write')
+                self.check_access_rule('write')
+                self._ensure_portal_user_can_write(vals.keys())
+                portal_can_create = True
```

Comment:
how do we know that task will be created in a project where the portal user can create tasks?

Reply:
The call to ``self.check_access_rule('create')``.

There was a typo 'write' instead of 'create'